doc/doc.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package doc
+
+// New returns a new document.
+// TODO: figure out if we can get away with fewer allocs, e.g. we could update
+// the signature to be New(id []byte, tagKeys [][]byte, tagValues [][]byte) Document {...}
```

Comment:
To that end, perhaps we can just take a slice of `Fields` and push it onto the caller to guarantee that the names are unique?

Reply:
That's definitely the intent. Don't want to take a slice of fields cause that's the whole point of this method. If they had that, they could create a `Document` anyway (it's exported after all).

Reply:
It seems more natural to me to take a slice of (key,value) pairs than a slice of keys and a slice of values. Perhaps we expose a `Field` type (I think there's one in the previous diff) and then we take a slice of them?

Reply:
Sure, all depends on how the original data is stored. E.g. if a user has a `[]Field` type, they can construct a document without using this ctor; the document type and it's fields are all exported. 

Lets come back and address this once we know what input data looks like on the m3db side.

Reply:
Maybe we can just make this `Fields` and then users can wrap it up however they want?

i.e.
```
type Fields interface {
  Iterator() FieldsIterator
}
```

Reply:
As discussed perhaps just remove the constructor for now?

Reply:
Yep, removed.